BitwardenShared/Core/Auth/Repositories/AuthRepository.swift

```diff
@@ -759,6 +767,11 @@ extension DefaultAuthRepository: AuthRepository {
         )
     }
 
+    func unlockVaultWithAuthenticatorVaultKey(userId: String) async throws {
+        let authenticatorKey = try await keychainService.getAuthenticatorVaultKey(userId: userId) ?? ""
+        try await unlockVault(method: .decryptedKey(decryptedUserKey: authenticatorKey), hadUserInteraction: false)
```

Comment:
🤔 If `keychainService.getAuthenticatorVaultKey` is `nil` then unlocking will definitely fail so it feels like a `guard` should be added here and throw an error if that happens so we have a more specific error as to why this failed.